### ISSUE TITLE
Invititations for events to Nextcloud users

### ISSUE THREAD

**Issue by alve89** [nextcloud/server#4299]
Hey guys,

here's a possible enhancement: Currently the creator of an event is able to invite people with their email address. A good point was to invite users or groups of the cloud instance (and / or with [external] Nextcloud instance mail addresses). I think it is not correctly to create a new calendar and share it with users / groups just for only one single event.

What do you think?

**Comment by MorrisJobke** [comment]
cc @georgehrke @jancborchardt

**Comment by georgehrke** [comment]
> here's a possible enhancement: Currently the creator of an event is able to invite people with their email address. A good point was to invite users or groups of the cloud instance (and / or with [external] Nextcloud instance mail addresses). I think it is not correctly to create a new calendar and share it with users / groups just for only one single event.

If users entered an email address in their personal setting, they will appear in the autocompletion when inviting attendees.

**Comment by alve89** [comment]
I tried this. The autocompletion doesn't uses the personal settings but the addressbook. But first it's not the same and second you can't suppose that every user did this. And if you could choose a user of the cloud instance the event was automatically added to his (cloud) calendar (as it works with the personal set mail address).

**Comment by georgehrke** [comment]
>  The autocompletion doesn't uses the personal settings but the users addressbook.

It searches both. There is a system address book with all the internal addresses. AFAIK the email address from personal settings is only synced periodically with the system address book, so you need to wait some time or use `php occ`

**Comment by Spartachetto** [comment]
If I understand properly what @alve89 was asking, the problem is that I can invite a Nextcloud user to the event only if s/he configured an email address.

I also have issues with this behaviour: if the person I want to invite is a user of the same instance of Nextcloud, why does s/he need an email address to get the information about the invite. Couldn't the server use a notification (@nickvergessen , maybe...)?

**Comment by skjnldsv** [comment]
@georgehrke accept or deny? :)

**Comment by georgehrke** [comment]
You can already invite Nextcloud user as long as they provide an email address in their profile.
(if they do, it will automatically be added to their calendar).

Regarding groups, there is also an existing ticket in the calendar repo

**Comment by alve89** [comment]
> 
> You can already invite Nextcloud user as long as they provide an email address in their profile.

And exactly that's the point: You can't be sure and you can't presuppose every user did this. So if a user didn't, it can't be invited. That's the point of my suggestion: Use user-names not (only) mail adresses.

**Comment by georgehrke** [comment]
> And exactly that's the point: You can't be sure and you can't presuppose every user did this. So if a user didn't, it can't be invited. That's the point of my suggestion: Use user-names not (only) mail adresses.

The entire CalDAV invitation system is build upon email addresses.

There are some ugly hacks that you can use (like using `principal:/...` addresses), but that will cause major issues regarding compatibility with other clients, especially when inviting both internal users and external users with their email address.

tl;dr: When you want to use calendar scheduling, make sure all users have their email address set.

**Comment by Spartachetto** [comment]
I deeply respect your willingness of respecting the standards.
Yet I still find counterintuitive Nextcloud behaviour: if I (a user of a Nextcloud instance) receive an invitation to an event from another user of the same Nextcloud instance, I would expect to receive a notification from the Nextcloud notification system. I do not know if @jancborchardt agrees, yet to me that would seem the most simple behaviour for all the involved users. I do get that, if there are users of other instances or invitees that are **not** Nextcloud users, the issue has to be managed differently. Yet the same instance issue seems to me so relevant that an ad hoc behaviour would be worthwhile

**Comment by georgehrke** [comment]
> I deeply respect your willingness of respecting the standards.

It is mostly about having a product that works with various clients.

> if I (a user of a Nextcloud instance) receive an invitation to an event from another user of the same Nextcloud instance, I would expect to receive a notification from the Nextcloud notification system.

This is not what this ticket is about. What you are requesting is https://github.com/nextcloud/calendar/issues/144. I'm thinking of a UI similar to https://www.busymac.com/images/bcl3mac-meeting-invitation-received.png.

**Comment by jancborchardt** [comment]
I do see the balance here – on one hand we want to honor the standards, on the other hand this is a papercut when people use our software.

I’d agree with what @Spartachetto wrote in their comment at https://github.com/nextcloud/server/issues/4299#issuecomment-521658878 – using the notifications system for people who didn’t set their email address.

**Comment by georgehrke** [comment]
> I’d agree with what @Spartachetto wrote in their comment at #4299 (comment) – using the notifications system for people who didn’t set their email address.

The point is that you can't invite them in the first place if they don't have an email address set. The entire invitation process is provided by Sabre/DAV and works solely based on the calendar-data we send to the server. The calendar-data contains [Attendee Properties](https://tools.ietf.org/html/rfc5545#section-3.8.4.1) with email-addresses as value.

If you don't know the email address, you can't send an invitation.
(Yes, you could use the `principal:/...` addresses for users without emails, but that's a really ugly hack that i'm not willing to implement.)

What we should solve instead imho is that users don't have email addresses set in their personal settings. The calendar app could perhaps send a notification on first login asking people to set up their email address in the settings?

**Comment by Spartachetto** [comment]
I do not have email addresses set in my personal settings because I do not want to.
There are practical reasons for that, and if you @georgehrke and @jancborchardt are interested I can describe my user case. The same is true for the other users of my family cloud. 
This means that I cannot use invitations in Nextcloud despite having my calendar in it and having the Nextcloud app on my mobile.
You were very clear and patient in explaining the technical reasons for it. I was trying to convey the user perception, which is "it doesn't work ".
At this point the evaluation is: am I a fringe user for not wanting to set an email address or are there enough users like me to let you consider to try to find a way to use Nextcloud notifications too?
I do get that finding a way could be technically challenging.
As I said I am available to describe my user case to help you both in the decision.

Thank you for your attention!

**Comment by georgehrke** [comment]
> As I said I am available to describe my user case to help you both in the decision.

Yes, i would be interested to know your reason. It is your own server, you are owning all the data.
Why would you not enter your email address into your personal settings?

**Comment by Spartachetto** [comment]
@georgehrke sorry to let you wait, I'll try to describe my reasons clearly. 

- **access to information**: I check my professional email several times a day, but now I realized that I check my personal email quite rarely (that means that sometimes some weeks can pass between a check and the other). This is true also because nowadays I receive mainly communication from providers (banks, electricity, ...) which are important but not urgent and a lot of communication / advertisement from hotel chains, airlines, ... . So picking up an inviation to an event would be not easy at all

- **privacy**: I still did not have enough time and willingness to leave one of the big internet mammoths as email provider. I'd love to avoid it to have access to all my invitations after I selected Nextcloud to have my agenda elsewhere

- **privacy 2**: I do not use federation just for lack of opportunity (no friend using Nextcloud) and I already set the sharing of my profile info for federation to the lower level possible, yet you know.... better safe than sorry :) Maybe I did not understand something, yet I do not want my email address to be available through that channel

Add to the previous reasons the ease of use for myself and the other users. For this last reason I'd love to receive the info that I got an invitation on my mobile and / or via the Nextcloud web interface (which I use often). Take also into account that I have the Nextcloud app (F-Droid version) on my mobile.

I was thinking that using Nextcloud notification system could be a good solution. I get the importance of :

> It is mostly about having a product that works with various clients.
